templates/common/infra/bicep/core/host/aks/appgw.bicep

```diff
@@ -0,0 +1,196 @@
+param resourceName string
+param location string
+param appGwSubnetId string
+param privateIpApplicationGateway string
+param availabilityZones array
+param userAssignedIdentity string
+param workspaceId string
+param appGWcount int
+param appGWmaxCount int
+
+var appgwName = 'agw-${resourceName}'
```

Comment:
We typically don't have prefix or naming in the modules, they should be passed in or referenced